spec/models/role_spec.rb

```diff
@@ -45,6 +45,13 @@
     end
   end
 
+  describe ".add_moderator" do
```

Comment:
I'd rather have a generic "add" method that takes :moderator as an argument. Otherwise you'll pretty quick bloat your models with repetitive methods like that.